src/views/SelectProduct.vue

```diff
@@ -222,20 +240,93 @@ export default defineComponent({
       products: 'product/getProducts',
       isScrollable: 'product/isScrollable',
       appliedFilters: 'product/getAppliedFilters',
-      query: 'product/getQuery'
+      query: 'product/getQuery',
+      pendingJobs: 'job/getPendingJobs',
+      getTagsAndOperator: 'job/getTagsAndOperator',
+      getCurrentEComStore:'user/getCurrentEComStore',
+      userProfile: 'user/getUserProfile',
+      currentEComStore: 'user/getCurrentEComStore',
+      shopifyConfig: 'util/getShopifyConfig',
+      facilitiesByProductStore: 'util/getFacilityByProductStore',
     })
   },
   data () {
     return {
       threshold: '' as any,
-      queryString: ''
+      queryString: '',
+      isFilterChanged: false,
+      isServiceScheduling: false,
+      job: {} as any,
+      jobId: "" as any
+    }
+  },
+  async ionViewWillEnter(){
+    if (this.$route.query.id) {
+      this.applyThresholdRule(this.$route.query.id)
     }
   },
   methods: {
+    async applyThresholdRule(id: any){
+      this.jobId = id;
+      let job = this.pendingJobs.find((job: any) => job.jobId === this.jobId)
+      job = job ? job : await JobService.fetchJob({eComStoreId: this.getCurrentEComStore.productStoreId, jobId: this.jobId})
+      if (job) {
+        this.job = job;
+        if (job.runtimeData?.searchPreferenceId) {
+          const includedTagsAndOperator = this.getTagsAndOperator(job.runtimeData.searchPreferenceId, "included");
```

Comment:
I think we should created a method that takes search preference and returns the appliedFilters object